src/main.cc

```diff
@@ -725,7 +725,10 @@ int run_server(StringView session, StringView server_init,
             }
         }
     }
-    catch (const kill_session&) {}
+    catch (const kill_session& e)
+    {
+        local_client_exit = e.status;
```

Comment:
There should be no need to do that as the local client will get removed as every other clients and will set that value already.